aip/general/0193.md

```diff
@@ -34,272 +34,154 @@ is necessary, you **should** provide a link where a reader can get more
-exactly one `ErrorInfo`. It is required.
+The reason should be terse, but meaningful enough for a human reader to
+understand what the reason refers to.
 
-[ErrorInfo Payload][ErrorInfo]
+Good examples:
 
-Details of this object are summarized in the following fields, field
-descriptions, and examples:
+- `CPU_AVAILABILITY`
+- `NO_STOCK`
+- `CHECKED_OUT`
+- `AVAILABILITY_ERROR`
 
-- `reason string`
-  - A short snake_case description of why the error occurred. Error
-    reasons are unique within a particular domain of errors. The error
-    reason **must** do the following:
-  - Be at most 63 characters and match a regular expression of
-    `/[A-Z][A-Z0-9_]+[A-Z0-9]/`, which represents UPPER_SNAKE_CASE.
-  - Be meaningful enough for a human reader to understand what the
-    reason refers to.
-  - Be unique and consumable by machine actors for automation.
-  - *Example*:  CPU_AVAILABILITY<br>
-    Distill your error message into its simplest form. For example, the
-    `reason string` could be one of the following text examples in
-    UPPER_SNAKE_CASE: `UNAVAILABLE`, `NO_STOCK`, `CHECKED_OUT`,
-    `AVAILABILITY_ERROR`, if your error message is,
-
-    > The Book, "The Great Gatsby", is unavailable at the Library,
-    > "Garfield East". It is expected to be available again on 2199-05-13.
-
-  - In contrast, using either of the following reasons is not
-    recommended: `THE_BOOK_YOU_WANT_IS_NOT_AVAILABLE`, `ERROR`. And,
-    using either of the following reasons breaches the required
-    formatting and is not allowed: `librariesAreGreat`, `noBooks`.
-
-- `domain string`
-  - The logical grouping to which the `reason` belongs. The error domain
-    is typically the registered service name of the tool or product that
-    generated the error. The domain must be a globally unique value.
-  - *Example*:<br>`pubsub.googleapis.com`
-
-- `metadata map`
-  - Additional structured details about this error, which **should**
-    provide important context for customers to identify resolution
-    steps.  Keys **should** match `/[a-z][a-zA-Z0-9-_]+/`, and be
-    limited to 64 characters in length. When identifying the current
-    value of an exceeded limit, the units **should** be contained in the
-    key, not the value.
-  - *Example*:
-    ```
-    "vmType": "e2-medium",
-    "attachment": "local-ssd=3,nvidia-t4=2",
-    "zone": "us-east1-a"
-    ```
-  - For guidance on using the metadata map, see [Dynamic
-    Variables](#dynamic-variables).
+Bad examples:
 
-##### Dynamic variables
+- `THE_BOOK_YOU_WANT_IS_NOT_AVAILABLE` (overly verbose)
+- `ERROR` (too general)
 
-The best, actionable error messages include dynamic segments. These
-variable parts of the message are specific to a particular request.
-Without such context, it is unlikely that the message will be fully
-actionable by the user.
+The `domain` field is the logical grouping to which the `reason` belongs.
+The error domain  typically the registered service name of the tool or product
```

Comment:
```suggestion
The error domain is typically the registered service name of the tool or product
```

Reply:
I hadn't even seen this sentence - I've somewhat repeated it later. Changed this whole paragraph.

Reply:
(I don't think "tool or product" is particularly useful here. Clients are talking to an API - what kind of "tool" is relevant?)